### PR TITLE
HIP-584: Add burn precompile

### PR DESCRIPTION
**Pull request by vtronkov**
Things TODO:
- add documentation on some of the file
- add tests
- fix a TODO in the code
- prepare the PR for "Ready for review"

**Description**:

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/SystemContractAbis.java [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+ * about; it is intended that ALL those other sources will be refactored away, leaving only this
+ * enum as the single source of information about these ABIs
+ * N.B.: Currently this file needs to be maintained by hand when ABIs are added or versioned; it is
+ * intended that this file be GENERATED from the actual Solidity contracts (in the smart contracts
+ * repository)
+ * N.B.: Need to add return types, and argument names (which could be used for various purposes)
+ */
+@SuppressWarnings("java:S1192") // duplicate string literals - in this case, removing duplication makes code
+//                       more difficult to read
+public enum SystemContractAbis {
```
[the remark this answers is not in the document]
Comment (0xivanov) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/SystemContractAbis.java:59]: Unless we want to keep these classes the same as services, but even there some methods are unused :)

**Review by IvanKavaldzhiev — COMMENTED** [review]
You need to GPG sign all your commits.

**Comment by vtronkov** [comment]
I made a copy branch and cherry-picked my commits from here and GPG singed them: https://github.com/hashgraph/hedera-mirror-node/pull/6290.
I will fix @0xivanov's comments in there. I'm closing the PR.

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/DecodingFacade.java [hunk 2]
```diff
@@ -42,21 +43,23 @@
 
 @Singleton
```
[the remark this answers is not in the document]
Comment (vtronkov) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/codec/DecodingFacade.java:44]: Fixed in https://github.com/hashgraph/hedera-mirror-node/pull/6290

File: hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/SystemContractAbis.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (vtronkov) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/SystemContractAbis.java:59]: I copied this class from https://github.com/hashgraph/hedera-mirror-node/pull/6277 so I'm guessing we want to keep it the same as the services. @IvanKavaldzhiev thoughts about that? Should I delete the unused constants and methods?

[the remark this answers is not in the document]
Comment (IvanKavaldzhiev) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/SystemContractAbis.java:59]: We would eventually need all of these constants, so we can copy them and keep them.

[the remark this answers is not in the document]
Comment (vtronkov) [reply hedera-mirror-web3/src/main/java/com/hedera/services/store/contracts/precompile/impl/SystemContractAbis.java:59]: Sounds good to me

**Pull request closed without merging**